iib/workers/tasks/opm_operations.py

```diff
@@ -738,3 +751,110 @@ def opm_create_empty_fbc(
         binary_image=binary_image,
         dockerfile_name='index.Dockerfile',
     )
+
+
+def opm_registry_add_fbc_fragment(
+    temp_dir: str,
+    from_index: str,
+    binary_image: str,
+    fbc_fragment: str,
+    overwrite_from_index_token: Optional[str],
+) -> None:
+    """
+    Add FBC fragment to from_index image.
+
+    This only produces the index.Dockerfile file and does not build the container image.
+
+    :param str temp_dir: the base directory to generate the database and index.Dockerfile in.
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from. This should point to a digest or stable tag.
+    :param str fbc_fragment: the fbc fragment to be added in from_index.
+    :pararm str overwrite_from_index_token: token used to access the image
+    """
+    fragment_path, fragment_operator = extract_fbc_fragment(temp_dir, fbc_fragment)
+
+    is_operator_in_db, index_db_path = verify_operator_exists(
+        from_index=from_index,
+        base_dir=temp_dir,
+        operator_package=fragment_operator,
+        overwrite_from_index_token=overwrite_from_index_token,
+    )
+
+    from_index_configs_dir = get_catalog_dir(from_index=from_index, base_dir=temp_dir)
+
+    if is_operator_in_db:
+        log.info('Removing %s from %s index.db ', fragment_operator, from_index)
+        _opm_registry_rm(index_db_path, [fragment_operator], temp_dir)
+        migated_catalog_dir, _ = opm_migrate(
```

Comment:
I am not sure but doesn't it rewrite the `from_index_configs_dir` data?

Reply:
I am thinking about if we do not migrate it at all? 
Because we just removed the operator? Or this is just to be sure that if anything else was changed it is copied over?

Reply:
the location of from_index_configs_dir will be like_/tmp/iib-1mnpn1yl/configs < this has the configs content of from_index
and the migrated catalog has its own dir, i.e /tmp/iib-1mnpn1yl/catalog < resulted from migrated_catalog

Reply:
yeah, even i thought we might skip the migration and one copy step , but I thought we should add it as a caution

Reply:
> 

This brings me the question what if the image has label `operators.operatorframework.io.index.configs.v1'` set 
 as `/catalog` . This might be legitimate and since we get the path from label I wold be cautious here.

Reply:
on the from_index, the label will always be set as we manage and create that index, and that's always /configs. But we never know what the future holds, so I can add change it to be safer on our end.

for the fbc_fragment, we still haven't got any response on the exact location of the catalog, hence I added a config in the worker so if it needs to change, we can change it easily. I believe the location should be same for all fbc_fragments

Reply:
Hey @lipoja , circling back on your question 

> "This brings me the question what if the image has label operators.operatorframework.io.index.configs.v1 set
> as /catalog . This might be legitimate and since we get the path from label I wold be cautious here."

Was your question for label in from_index or fbc_fragment? For fbc_fragment we don't know what will be label, and irrespective of the location, it can be specified in configs in https://github.com/release-engineering/iib/pull/484/files#diff-ca2cc097d7757c5b9d5ef4de2c4ed7d90e1431835c57c042a50d4cb8b35994c2R77 and even if the label or "config or catalog" it will not matter as the dir is copied to /tmp/iib-**/**fbc-fragment**

I was also thinking that it is very unlikely that the label will be set as "/catalog" for from_index as if that's done even the normal add request will not work, it will fail during opm_migrate or opm_migrate will overwrite the catalog folder

Reply:
Thanks

Reply:
The question was for label for fbc_fragment directory. Yeah lets keep it this way - I agree